src/Fixer/Casing/FloatLiteralCaseFixer.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Casing;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+final class FloatLiteralCaseFixer extends AbstractFixer
+{
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Float literals must be in correct case.',
+            [
+                new CodeSample(
+                    "<?php\n\$foo = 1e80;\n\$bar = 2.5e-3;\n"
+                ),
+            ]
+        );
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isTokenKindFound(T_DNUMBER);
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        foreach ($tokens as $index => $token) {
+            if (!$token->isGivenKind(T_DNUMBER)) {
+                continue;
+            }
+
+            $content = $token->getContent();
+
+            $newContent = str_replace('e', 'E', $content);
```

Comment:
should the normalized case be `e` or `E`?

`E` is used by default by php in float dumps

`e` (LC) seems a little more readable and consistent with `0x`

Reply:
reading your reasoning, I would choose the default based on PHP default, not other language default.

yet also `dumps` often provide a weird format. do we have other examples from PHP ecosystem, by any chance?

Reply:
PHP - basically any cast from float to string: https://3v4l.org/qhj8J